test/e2e/usersignup_test.go

```diff
@@ -45,662 +43,190 @@ func (s *userSignupIntegrationTest) TearDownTest() {
-				Reason: "PendingApproval",
-			}))
-		require.NoError(s.T(), err)
-
-		// Now, manually approve it (setting Approved to true) and update the resource
-		userSignup.Spec.Approved = true
-
-		err = s.awaitility.Client.Update(context.TODO(), userSignup)
-		require.NoError(s.T(), err)
-
-		// Lookup the UserSignup again
-		userSignup, err = s.hostAwait.WaitForUserSignup(userSignup.Name)
-		require.NoError(s.T(), err)
+		// then
+		s.checkUserSignupManualApproval()
+	})
 
-		require.NotEmpty(s.T(), userSignup.Status.CompliantUsername)
-
-		// Confirm that the conditions are updated to reflect that the userSignup was approved
-		_, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(
-			v1alpha1.Condition{
-				Type:   v1alpha1.UserSignupApproved,
-				Status: corev1.ConditionTrue,
-				Reason: "ApprovedByAdmin",
-			},
-			v1alpha1.Condition{
-				Type:   v1alpha1.UserSignupComplete,
-				Status: corev1.ConditionTrue,
-			}))
-		require.NoError(s.T(), err)
+	s.T().Run("manual approval policy", func(t *testing.T) {
+		// when
+		s.setApprovalPolicyConfig("manual")
 
-		// Confirm the MUR was created
-		s.assertCreatedMUR(userSignup)
+		// then
+		s.checkUserSignupManualApproval()
 	})
 
-	s.T().Run("usersignup created with approved set to true", func(t *testing.T) {
-		// Create user signup - approval set to true
-		s.T().Logf("Creating UserSignup with namespace %s", s.namespace)
-		userSignup, err := newUserSignup(s.awaitility.Host(), uuid.NewV4().String(), "robertjones@magenta.com")
-		require.NoError(s.T(), err)
+	s.T().Run("auto approval policy", func(t *testing.T) {
+		s.setApprovalPolicyConfig("automatic")
 
-		userSignup.Spec.Approved = true
-		err = s.awaitility.Client.Create(context.TODO(), userSignup, testsupport.CleanupOptions(s.testCtx))
-		require.NoError(s.T(), err)
-		s.T().Logf("user signup '%s' created", userSignup.Name)
-
-		// Confirm that:
-		// 1) Confirm the UserSignup was created
-		// 2) the Approved condition is set to true
-		// 3) the Approved reason is set to ApprovedByAdmin
-		// 4) the Complete condition is set to true
-		userSignup, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(
-			v1alpha1.Condition{
-				Type:   v1alpha1.UserSignupApproved,
-				Status: corev1.ConditionTrue,
-				Reason: "ApprovedByAdmin",
-			},
-			v1alpha1.Condition{
-				Type:   v1alpha1.UserSignupComplete,
-				Status: corev1.ConditionTrue,
-			}))
-		require.NoError(s.T(), err)
+		s.T().Run("approved set to false", func(t *testing.T) {
+			// UserSignup created with approved flag set to false but it should be approved anyway
+			s.createUserSignupAndAssertAutoApproval(false)
+		})
 
-		// Confirm the MUR was created
-		s.assertCreatedMUR(userSignup)
+		s.T().Run("approved set to true", func(t *testing.T) {
+			// UserSignup created with approved flag set to true and it should be approved
+			s.createUserSignupAndAssertManualApproval(true)
```

Comment:
```suggestion
			s.createUserSignupAndAssertAutoApproval(true)
```

Reply:
No, no. In this case we will be using "manual" approval anyway. The `Spec.Approved` set to true. So, it doesn't matter if auto-approval is enabled or not. So, the status reason will be set to "ApprovedByAdmin". Not "ApprovedAutomatically". So, the check is correct here.

Reply:
ah, I see. It was very confusing that it's inside of the `"auto approval policy"` subtests, but using checks for manual approval - I didn't notice the different resulting status...

Reply:
Let me maybe add a comment to the test so it's clear why we still assert "manual" approval in auto-approval test :)

Reply:
btw, could we merge the three functions:
`createUserSignupAndAssertManualApproval`
`createUserSignupWithNameAndAssertManualApproval`
`createUserSignupAndAssertAutoApproval`
into one:
```go
func (s *userSignupIntegrationTest) createAndCheckUserSignup(specApproved bool, conditions ...v1alpha1.Condition) *v1alpha1.UserSignup {
	// Create a new UserSignup with the given approved flag
	userSignup := newUserSignup(s.T(), s.awaitility.Host(), uuid.NewV4().String(), "testuser"+uuid.NewV4().String())
	userSignup.Spec.Approved = specApproved
	err := s.awaitility.Client.Create(context.TODO(), userSignup, testsupport.CleanupOptions(s.testCtx))
	require.NoError(s.T(), err)
	s.T().Logf("user signup '%s' created", userSignup.Name)

	// Check the UserSignup is approved now
	userSignup, err = s.hostAwait.WaitForUserSignup(userSignup.Name, wait.UntilUserSignupHasConditions(conditions...))
	require.NoError(s.T(), err)

	return userSignup
}
```
and then use it:
```go
s.T().Run("approved set to false", func(t *testing.T) {
	// UserSignup created with approved flag set to false but it should be approved anyway
	userSignup := s.createAndCheckUserSignup(false, approvedAutomatically()...)
	// Confirm the MUR was created
	s.assertCreatedMUR(userSignup)
})

s.T().Run("approved set to true", func(t *testing.T) {
	// UserSignup created with approved flag set to true and it should be approved
	userSignup := s.createAndCheckUserSignup(true, approvedByAdmin()...)
	// Confirm the MUR was created
	s.assertCreatedMUR(userSignup)
})
```
and in the `createUserSignupAndAssertPendingApproval` in this way:
```go
func (s *userSignupIntegrationTest) createUserSignupAndAssertPendingApproval() *v1alpha1.UserSignup {
	// Create a new UserSignup with approved flag set to false
	userSignup := s.createAndCheckUserSignup(false, pendingApproval()...)

	// Confirm the CompliantUsername has NOT been set
	require.Empty(s.T(), userSignup.Status.CompliantUsername)

	// Confirm that a MasterUserRecord wasn't created
	_, err := s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second * 10)).WaitForMasterUserRecord(userSignup.Spec.Username)
	require.Error(s.T(), err)
	return userSignup
}
```
it would make the code shorter.

Reply:
We can't auto-generate the username always (we need to use specific ones in username transformation tests) but I made the code shorter: https://github.com/codeready-toolchain/toolchain-e2e/pull/47/commits/381ca0d949494b7a7872c0e82539637b31fd5cbb

Reply:
Sure, when I was writing it I hoped that we could drop the transformation tests. Anyway, I see that we need it so passing the username is just one parameter in the methods. 
I still believe that we could consolidate it more according to my proposal (drop some methods and make `createUserSignupAndAssertPendingApproval` shorter), but your changes are sufficient.